src/crystal/dwarf/strings.cr

```diff
@@ -4,7 +4,7 @@ module Crystal
       def initialize(@io : IO::FileDescriptor, @offset : UInt32 | UInt64, size)
         # Read a good chunk of bytes to decode strings faster
         # (avoid seeking/reading the IO too many times)
-        @buffer = Bytes.new(Math.max(16384, size))
+        @buffer = Bytes.new(Math.min(Math.max(16384, size), io.info.size - offset))
```

Comment:
**question:** I'm a bit puzzled by this change. Why is it necessary?

Reply:
`gcc` does not guarantee any particular ELF section order; it happens that `.debug_line_str` could end up being less than 16 KB away from the end of the file, and the `read_fully` below fails without this change. This of course could happen on other systems too but I have never observed this outside OmniOS. Maybe `read_fully` could be replaced with `read` if caching those strings is not critical